Content.Server/_Impstation/CosmicCult/CosmicCultSystem.cs

```diff
@@ -0,0 +1,279 @@
+            {
+                comp.CultistsCheckTimer = _timing.CurTime + comp.CheckWait;
+                var cultistsPresent = CultistCount = _cosmicGlyphs.GatherCultists(uid, 5).Count; //Let's use the cultist collecting hashset from Cosmic Glyphs to see how many folks are around!
+                CultistCount = int.Clamp(cultistsPresent, 0, 10);
+                _popup.PopupCoordinates(Loc.GetString("cosmiccult-finale-cultist-count", ("COUNT", CultistCount)), Transform(uid).Coordinates);
+                var modifyTime = TimeSpan.FromSeconds(420 * 5 / (420 - 36 * CultistCount) - 5);
+                comp.BufferTimer -= modifyTime;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Parses marker components to output their respective loc strings directly into your examine box, courtesy of TGRCdev(Github).
+    /// </summary>
+    private void MakeSimpleExamineHandler<TComp>(LocId message)
+    where TComp: IComponent
+    {
+        SubscribeLocalEvent((Entity<TComp> ent, ref ExaminedEvent args) => {
+            if (HasComp<CosmicCultComponent>(args.Examiner))
+                args.PushMarkup(Loc.GetString("cosmic-examine-text-forthecult"));
+            else
+                args.PushMarkup(Loc.GetString(message, ("entity", ent.Owner)));
+        });
+    }
+    #endregion
+
+    #region Init Cult
+    /// <summary>
+    /// Add the starting powers to the cultist.
+    /// </summary>
+    private void OnStartCultist(Entity<CosmicCultComponent> uid, ref ComponentInit args)
+    {
+        foreach (var actionId in uid.Comp.CosmicCultActions)
+        {
+            var actionEnt = _actions.AddAction(uid, actionId);
+            uid.Comp.ActionEntities.Add(actionEnt);
+        }
+        if (TryComp<EyeComponent>(uid, out var eye))
+            _eye.SetVisibilityMask(uid, eye.VisibilityMask | MonumentComponent.LayerMask);
+    }
+
+    /// <summary>
+    /// Add the Monument summon action to the cult lead.
+    /// </summary>
+    private void OnStartCultLead(Entity<CosmicCultLeadComponent> uid, ref ComponentInit args)
+    {
+        _actions.AddAction(uid, ref uid.Comp.CosmicMonumentActionEntity, uid.Comp.CosmicMonumentAction, uid);
+    }
+
+    /// <summary>
+    /// Called by Cosmic Siphon. Increments the Cult's global objective tracker.
+    /// </summary>
+    #endregion
+
+    #region Entropy
+    private void OnStartMonument(Entity<MonumentComponent> uid, ref ComponentInit args)
+    {
+        _cultRule.MonumentTier1(uid);
+        _cultRule.UpdateCultData(uid);
+    }
+
+    private void OnInfuseEntropy(Entity<MonumentComponent> uid, ref InteractUsingEvent args)
+    {
+        if (!HasComp<CosmicEntropyMoteComponent>(args.Used) || !HasComp<CosmicCultComponent>(args.User) || !uid.Comp.Enabled || args.Handled)
+            return;
+        args.Handled = AddEntropy(uid, args.Used, args.User);
+    }
+    private bool AddEntropy(Entity<MonumentComponent> monument, EntityUid entropy, EntityUid cultist)
```

Comment:
Adding entropy does not update the UI's Entropy count without reloading, fix that

Reply:
Dunno but fix it :Godo:
Also I just realized there's a missing newline at this function :)

Reply:
More realistically, make sure that updating the UI state informs the client about it.
Compare with e.g. NanoChat or Vendomat UIs, which update on user input